src/components/yeti/FeaturesList.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+  <ul class="FeaturesList">
+    <li
+      v-for="(feature, i) of features"
+      :key="i"
+      class="FeaturesList-Item tag"
+      @mouseenter="showFeature(feature)"
+      @mouseleave="hideFeature(feature)"
+    >
+      <span>{{ nbPointsOnFeature(feature) }} points</span>
+      <button class="delete is-small" @click="removeFeature(feature)">x</button>
+    </li>
+  </ul>
+</template>
+
+<script>
+export default {
+  props: {
+    map: {
```

Comment:
Having big object sent as props to components is an anti-pattern : all its property will be decorated to allow the component to re-render, and it can make the UI terribly slow.

A better pattern would be to modify   `FeatureList` by emitting events. For instance, on line 11 : 

```
@click="$emit('removeFeature', feature)"
```

And on parent component

```
@remove-feature="map.removeFeature"
```

With that, you can keep the map object internal to the parent component.

Same principle with features object (even if they are smaller that map object)

Reply:
The point here is that the `map` object is coming from two level up. So I had to emit an event to the parent, then emit again to the grand-parent. It can quickly become overskill. That's why I've chosen to pass `map` as `props` instead.

First question: do you know a way to measure performance impact of both solutions?

Then, which solution is better: emitting events bubbling from parent to grand-parent? or using an EventBus? or another solution?

Reply:
I am not fluent enough with vue to decide whether an event bus is appropriate for c2c app because of its complexity, but I assume it's ok in this state, that's what I would rather use.

Reply:
I couldn't find any use of `EventBus` anywhere in the code, so I'm not in favor of adding it in this case. So, two solutions remain:

- left untouched. I'm not sure there is a perf issue here. Components are already under Vue's radar, passing them through props is only a reference.
- use events to send to parent, then grand-parent. It will break if adding another level of nesting one day.